runtime/criusupport/criusupport.cpp

```diff
@@ -54,46 +51,88 @@ setupJNIFieldIDs(JNIEnv *env)
 	jclass criuSystemCheckpointExceptionClass = NULL;
 	jclass criuJVMRestoreExceptionClass = NULL;
 	jclass criuSystemRestoreExceptionClass = NULL;
+	bool returnCode = true;
+	PORT_ACCESS_FROM_VMC(currentThread);
+	IDATA libCRIUReturnCode = 0;
 
 	UT_MODULE_LOADED(J9_UTINTERFACE_FROM_VM(vm));
 
 	criuJVMCheckpointExceptionClass = env->FindClass("org/eclipse/openj9/criu/JVMCheckpointException");
 	Assert_CRIU_notNull(criuJVMCheckpointExceptionClass);
-	vm->criuJVMCheckpointExceptionClass = (jclass) env->NewGlobalRef(criuJVMCheckpointExceptionClass);
+	vm->checkpointState.criuJVMCheckpointExceptionClass = (jclass) env->NewGlobalRef(criuJVMCheckpointExceptionClass);
```

Comment:
`NewGlobalRef` can fail, so there should probably be asserts.

Reply:
I see checks below, so why are the method IDs asserted? Should probably be consistent.

Reply:
Note `GetMethodID` sets an exception on failure as well.

Reply:
Actually `NewGlobalRef` asserts on failure, so maybe this code is fine as it is.